EXC_BAD_ACCESS Attempted to dereference garbage pointer

OS:
- [ ]  Windows
- [x]  MacOS
- [ ]  Linux

_Platform:_
- [x]  iOS
- [ ]  Android

Output of `node -v && npm -v && npm ls --prod --depth=0`

```
$ node -v && npm -v && npm ls --prod --depth=0
v8.5.0
5.4.2
SCApp@0.0.1 /Users/ryan/p/SCApp
├── accounting@0.4.1
├── async-retry@1.1.2
├── code-push-cli@2.1.0-beta
├── common-tags@1.4.0
├── credit-card-type@5.0.4
├── gaussian@1.1.0
├── humps@2.0.1
├── invariant@2.2.2
├── js-search@1.4.2
├── lodash@4.17.4
├── moment@2.18.1
├── path@0.12.7
├── plist@2.1.0
├── prettier@1.7.0
├── prop-types@15.6.0
├── UNMET PEER DEPENDENCY react@16.0.0-alpha.12
├── react-mixin@4.0.0
├── react-native@0.48.4
├── react-native-adjust@4.11.5
├── react-native-branch@2.0.0
├── react-native-camera@0.10.0
├── react-native-code-push@5.1.1-beta
├── react-native-deprecated-custom-components@0.1.1
├── react-native-fbsdk@0.6.1 invalid
├── react-native-fs@2.5.2
├── react-native-keychain@2.0.0-rc
├── react-native-progress@3.4.0
├── react-native-sentry@0.24.2
├── react-native-sound@0.10.4
├── react-navigation@1.0.0-beta.12
├── react-redux@5.0.6
├── redux@3.7.2
├── redux-logger@3.0.6
├── redux-thunk@2.2.0
└── validate.js@0.11.1

npm ERR! peer dep missing: react@0.13.x || 0.14.x || ^15.0.0-0 || 15.x, required by enzyme@2.9.1
npm ERR! invalid: react-native-fbsdk@0.6.1 /Users/ryan/p/SCApp/node_modules/react-native-fbsdk
```

Config:

```
 Sentry.config('https://e4eed1357f3948eb8e0c89bfc2d425d7:d15d972ebc6b4b27b7e300efec2863cc@sentry.io/76952', {
    deactivateStacktraceMerging: true,
    disableNativeIntegration: false,
  }).install();
```



---
I have following issue:
After applying a code-push update and restarting (using `codePush.restartApp()`), the app immediately crashes. CodePush rolls the app back to the previous build. `deactivateStacktraceMerging` is disabled because it was throwing from exceeding the max call stack size. 

It's also worth noting that if `disableNativeIntegration` is enabled, the app will no longer crash on launch after CodePush update, but intentional crashes will not be sent to Sentry.

A crash will also not occur if Sentry is never `install()`'ed.

**Steps to reproduce:**
1. Archive and release build on iTunes Connect.
1. Perform CodePush release (with mandatory release at startup).
1. Install release build from iTunes Connect.
1. Launch app.

**Actual result:**
1. App launches, downloads and installs CodePush update.
1. App automatically restarts with new CodePush update

**Expected result:**
1. App launches, downloads and installs CodePush update.
1. App restarts and immediately crashes


Stack Trace:
```
OS Version: iOS 10.3.3 (14G60)
Report Version: 104

Exception Type: EXC_BAD_ACCESS (SIGBUS)
Exception Codes: BUS_NOOP at 0x000000010000078c
Crashed Thread: 11

Application Specific Information:
Attempted to dereference garbage pointer 0x10000078c.
Originated at or in a subcall of RCTFBQuickPerformanceLoggerConfigureHooks

Thread 0 name: 
0   libsystem_kernel.dylib          0x305609260         semaphore_wait_trap
1   libdispatch.dylib               0x3053d25e8         _os_semaphore_wait
2   libdispatch.dylib               0x3053d1ca0         _dispatch_semaphore_wait_slow
3   CoreMotion                      0x31564d398         CLStartStopAdvertisingBeacon
4   CoreMotion                      0x315625440         CLStartStopAdvertisingBeacon
5   S. Contacts                     0x2001fb488         -[RCTSensorOrientationChecker init] (RCTSensorOrientationChecker.m:27)
6   S. Contacts                     0x2001f4c4c         -[RCTCameraManager init] (RCTCameraManager.m:318)
7   S. Contacts                     0x20013b7e8         __44-[RCTModuleData initWithModuleClass:bridge:]_block_invoke (RCTModuleData.mm:62)
8   S. Contacts                     0x20013bb24         -[RCTModuleData setUpInstanceAndBridge] (RCTModuleData.mm:109)
9   S. Contacts                     0x200146864         RCTUnsafeExecuteOnMainQueueSync (RCTUtils.m:259)
10  S. Contacts                     0x20013c3b8         -[RCTModuleData instance] (RCTModuleData.mm:237)
11  S. Contacts                     0x200101db0         __49-[RCTCxxBridge _prepareModulesWithDispatchGroup:]_block_invoke (RCTCxxBridge.mm:671)
12  libdispatch.dylib               0x3053bf9e0         _dispatch_call_block_and_release
13  libdispatch.dylib               0x3053bf9a0         _dispatch_client_callout
14  libdispatch.dylib               0x3053c43c0         _dispatch_main_queue_callback_4CF
15  CoreFoundation                  0x3074d30c8         __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
16  CoreFoundation                  0x3074d0ce4         __CFRunLoopRun
17  CoreFoundation                  0x307400da4         CFRunLoopRunSpecific
18  GraphicsServices                0x30a8d2074         GSEventRunModal
19  UIKit                           0x31390ac9c         UIApplicationMain
20  S. Contacts                     0x20007ab3c         main (main.m:16)
21  libdyld.dylib                   0x30542259c         start

Thread 1 name: 
0   libsystem_kernel.dylib          0x305627a88         __workq_kernreturn
1   libsystem_pthread.dylib         0x3057d11a4         _pthread_wqthread

Thread 2 name: 
0   libsystem_kernel.dylib          0x305627a88         __workq_kernreturn
1   libsystem_pthread.dylib         0x3057d11a4         _pthread_wqthread

Thread 3 name: 
0   libsystem_kernel.dylib          0x305627a88         __workq_kernreturn
1   libsystem_pthread.dylib         0x3057d0fd0         _pthread_wqthread

Thread 4 name: com.apple.uikit.eventfetch-thread
0   libsystem_kernel.dylib          0x305609224         mach_msg_trap
1   libsystem_kernel.dylib          0x30560909c         mach_msg
2   CoreFoundation                  0x3074d2e90         __CFRunLoopServiceMachPort
3   CoreFoundation                  0x3074d0ae4         __CFRunLoopRun
4   CoreFoundation                  0x307400da4         CFRunLoopRunSpecific
5   Foundation                      0x308a32db4         -[NSRunLoop(NSRunLoop) runMode:beforeDate:]
6   Foundation                      0x308a53b84         -[NSRunLoop(NSRunLoop) runUntilDate:]
7   UIKit                           0x314294830         -[UIEventFetcher threadMain]
8   Foundation                      0x308b30318         __NSThread__start__
9   libsystem_pthread.dylib         0x3057d368c         _pthread_body
10  libsystem_pthread.dylib         0x3057d359c         _pthread_start

Thread 5 name: 
0   libsystem_kernel.dylib          0x305627a88         __workq_kernreturn
1   libsystem_pthread.dylib         0x3057d11a4         _pthread_wqthread

Thread 6 name: com.facebook.react.JavaScript
0   libsystem_kernel.dylib          0x305609260         semaphore_wait_trap
1   libdispatch.dylib               0x3053d25e8         _os_semaphore_wait
2   libdispatch.dylib               0x3053d1ca0         _dispatch_semaphore_wait_slow
3   CoreMotion                      0x31564d398         CLStartStopAdvertisingBeacon
4   CoreMotion                      0x3156256e4         CLStartStopAdvertisingBeacon
5   libobjc.A.dylib                 0x304b0af00         object_cxxDestructFromClass(objc_object*, objc_class*)
6   libobjc.A.dylib                 0x304b18334         objc_destructInstance
7   libobjc.A.dylib                 0x304b18398         object_dispose
8   S. Contacts                     0x2001fb588         -[RCTSensorOrientationChecker dealloc] (RCTSensorOrientationChecker.m:38)
9   S. Contacts                     0x2001fa330         -[RCTCameraManager .cxx_destruct] (RCTCameraManager.m:21)
10  libobjc.A.dylib                 0x304b0af00         object_cxxDestructFromClass(objc_object*, objc_class*)
11  libobjc.A.dylib                 0x304b18334         objc_destructInstance
12  libobjc.A.dylib                 0x304b18398         object_dispose
13  S. Contacts                     0x20013d0a0         -[RCTModuleData .cxx_destruct] (RCTModuleData.mm:22)
14  libobjc.A.dylib                 0x304b0af00         object_cxxDestructFromClass(objc_object*, objc_class*)
15  libobjc.A.dylib                 0x304b18334         objc_destructInstance
16  libobjc.A.dylib                 0x304b18398         object_dispose
17  CoreFoundation                  0x307404a80         -[__NSArrayI dealloc]
18  S. Contacts                     0x2001031b8         __26-[RCTCxxBridge invalidate]_block_invoke (RCTCxxBridge.mm:923)
19  S. Contacts                     0x200126768         facebook::react::tryAndReturnError(std::__1::function<void ()> const&) (RCTCxxUtils.mm:97)
20  S. Contacts                     0x2000ff068         -[RCTCxxBridge _tryAndHandleError:] (RCTCxxBridge.mm:227)
21  Foundation                      0x308b3050c         __NSThreadPerformPerform
22  CoreFoundation                  0x3074d342c         __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
23  CoreFoundation                  0x3074d2d9c         __CFRunLoopDoSources0
24  CoreFoundation                  0x3074d09a8         __CFRunLoopRun
25  CoreFoundation                  0x307400da4         CFRunLoopRunSpecific
26  S. Contacts                     0x2000fefa8         +[RCTCxxBridge runRunLoop] (RCTCxxBridge.mm:219)
27  Foundation                      0x308b30318         __NSThread__start__
28  libsystem_pthread.dylib         0x3057d368c         _pthread_body
29  libsystem_pthread.dylib         0x3057d359c         _pthread_start

Thread 7 name: 
0   libsystem_kernel.dylib          0x305627314         __semwait_signal
1   libsystem_c.dylib               0x3054788b0         nanosleep
2   libc++.1.dylib                  0x304a5f3c4         std::__1::this_thread::sleep_for(std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > const&)
3   JavaScriptCore                  0x30fdb6c7c         bmalloc::Heap::scavenge(std::__1::unique_lock<bmalloc::StaticMutex>&, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000l> >)
4   JavaScriptCore                  0x30fdb6a50         bmalloc::Heap::concurrentScavenge()
5   JavaScriptCore                  0x30fdb7e30         bmalloc::AsyncTask<bmalloc::Heap, void (bmalloc::Heap::*)()>::threadRunLoop()
6   JavaScriptCore                  0x30fdb7d5c         bmalloc::AsyncTask<bmalloc::Heap, void (bmalloc::Heap::*)()>::threadEntryPoint(bmalloc::AsyncTask<bmalloc::Heap, void (bmalloc::Heap::*)()>*)
7   JavaScriptCore                  0x30fdb800c         void* std::__1::__thread_proxy<std::__1::tuple<void (*)(bmalloc::AsyncTask<bmalloc::Heap, void (bmalloc::Heap::*)()>*), bmalloc::AsyncTask<bmalloc::Heap, void (bmalloc::Heap::*)()>*> >(void*)
8   libsystem_pthread.dylib         0x3057d368c         _pthread_body
9   libsystem_pthread.dylib         0x3057d359c         _pthread_start

Thread 8 name: 
0   libsystem_kernel.dylib          0x305627a88         __workq_kernreturn
1   libsystem_pthread.dylib         0x3057d11a4         _pthread_wqthread

Thread 9 name: com.apple.NSURLConnectionLoader
0   libsystem_kernel.dylib          0x305609224         mach_msg_trap
1   libsystem_kernel.dylib          0x30560909c         mach_msg
2   CoreFoundation                  0x3074d2e90         __CFRunLoopServiceMachPort
3   CoreFoundation                  0x3074d0ae4         __CFRunLoopRun
4   CoreFoundation                  0x307400da4         CFRunLoopRunSpecific
5   CFNetwork                       0x308341df4         +[NSURLConnection(Loader) _resourceLoadLoop:]
6   Foundation                      0x308b30318         __NSThread__start__
7   libsystem_pthread.dylib         0x3057d368c         _pthread_body
8   libsystem_pthread.dylib         0x3057d359c         _pthread_start

Thread 10 name: com.apple.CoreMotion.MotionThread
0   libsystem_kernel.dylib          0x305609224         mach_msg_trap
1   libsystem_kernel.dylib          0x30560909c         mach_msg
2   IOKit                           0x307b5a1ac         io_hideventsystem_open
3   IOKit                           0x307b4bcf8         __IOHIDEventSystemClientRefresh
4   IOKit                           0x307b4c864         IOHIDEventSystemClientCreateWithType
5   CoreMotion                      0x31563a208         CLStartStopAdvertisingBeacon
6   CoreMotion                      0x31563af4c         CLStartStopAdvertisingBeacon
7   CoreMotion                      0x315645ae8         CLStartStopAdvertisingBeacon
8   CoreMotion                      0x315645a14         CLStartStopAdvertisingBeacon
9   CoreMotion                      0x31564d3c4         CLStartStopAdvertisingBeacon
10  CoreMotion                      0x3156459ac         CLStartStopAdvertisingBeacon
11  CoreMotion                      0x31568f7d8         CLMotionActivity::isSimilar(CLMotionActivity const&, CLMotionActivity const&)
12  CoreMotion                      0x3155ecc70         <redacted>
13  CoreMotion                      0x31568f550         CLMotionActivity::isSimilar(CLMotionActivity const&, CLMotionActivity const&)
14  libdispatch.dylib               0x3053bf9a0         _dispatch_client_callout
15  libdispatch.dylib               0x3053c06cc         dispatch_once_f
16  CoreMotion                      0x31568f5b8         CLMotionActivity::isSimilar(CLMotionActivity const&, CLMotionActivity const&)
17  CoreMotion                      0x3156253a0         CLStartStopAdvertisingBeacon
18  CoreMotion                      0x315626c54         CLStartStopAdvertisingBeacon
19  CoreMotion                      0x3156276c4         CLStartStopAdvertisingBeacon
20  CoreMotion                      0x315625844         CLStartStopAdvertisingBeacon
21  CoreMotion                      0x31564d670         CLStartStopAdvertisingBeacon
22  CoreMotion                      0x31564d710         CLStartStopAdvertisingBeacon
23  CoreFoundation                  0x3074d330c         __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__
24  CoreFoundation                  0x3074d2b28         __CFRunLoopDoBlocks
25  CoreFoundation                  0x3074d0e1c         __CFRunLoopRun
26  CoreFoundation                  0x307400da4         CFRunLoopRunSpecific
27  CoreFoundation                  0x30744d9c0         CFRunLoopRun
28  CoreMotion                      0x31564cfdc         CLStartStopAdvertisingBeacon
29  libsystem_pthread.dylib         0x3057d368c         _pthread_body
30  libsystem_pthread.dylib         0x3057d359c         _pthread_start

Thread 12 name: 
0   libsystem_kernel.dylib          0x305627a88         __workq_kernreturn
1   libsystem_pthread.dylib         0x3057d11a4         _pthread_wqthread

Thread 13 name: WebThread
0   libsystem_kernel.dylib          0x305609224         mach_msg_trap
1   libsystem_kernel.dylib          0x30560909c         mach_msg
2   CoreFoundation                  0x3074d2e90         __CFRunLoopServiceMachPort
3   CoreFoundation                  0x3074d0ae4         __CFRunLoopRun
4   CoreFoundation                  0x307400da4         CFRunLoopRunSpecific
5   WebCore                         0x310d6ad3c         RunWebThread(void*)
6   libsystem_pthread.dylib         0x3057d368c         _pthread_body
7   libsystem_pthread.dylib         0x3057d359c         _pthread_start

Thread 14 name: 
0   libsystem_kernel.dylib          0x305627314         __semwait_signal
1   libsystem_c.dylib               0x3054788b0         nanosleep
2   libsystem_c.dylib               0x3054e1270         sleep
3   S. Contacts                     0x20028fdbc         monitorCachedData (KSCrashCachedData.c:148)
4   libsystem_pthread.dylib         0x3057d368c         _pthread_body
5   libsystem_pthread.dylib         0x3057d359c         _pthread_start

Thread 15 name: KSCrash Exception Handler (Secondary)
0   libsystem_kernel.dylib          0x305609224         mach_msg_trap
1   libsystem_kernel.dylib          0x30560909c         mach_msg
2   S. Contacts                     0x20026c024         handleExceptions (KSCrashMonitor_MachException.c:281)
3   libsystem_pthread.dylib         0x3057d368c         _pthread_body
4   libsystem_pthread.dylib         0x3057d359c         _pthread_start

Thread 17 name: 
0   libsystem_kernel.dylib          0x305627a88         __workq_kernreturn
1   libsystem_pthread.dylib         0x3057d11a4         _pthread_wqthread

Thread 18 name: 
0   libsystem_kernel.dylib          0x305609314         swtch_pri
1   libsystem_pthread.dylib         0x3057d3b8c         _pthread_find_thread
2   libsystem_pthread.dylib         0x3057d40b8         _pthread_lookup_thread
3   libsystem_pthread.dylib         0x3057d3f68         pthread_detach
4   JavaScriptCore                  0x30fdae5e4         WTF::detachThread(unsigned int)
5   JavaScriptCore                  0x30fd8dc4c         WTF::AutomaticThreadCondition::notifyOne(WTF::Locker<WTF::LockBase> const&)
6   JavaScriptCore                  0x30fa15fe0         JSC::Heap::requestCollection(std::optional<JSC::CollectionScope>)
7   JavaScriptCore                  0x30fa15ea4         JSC::Heap::collectAsync(std::optional<JSC::CollectionScope>)
8   JavaScriptCore                  0x30fa06cb0         JSC::FullGCActivityCallback::doCollection()
9   JavaScriptCore                  0x30fa0d838         JSC::GCActivityCallback::doWork()
10  JavaScriptCore                  0x30f40e900         JSC::HeapTimer::timerDidFire(__CFRunLoopTimer*, void*)
11  CoreFoundation                  0x3074d3aa4         __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__
12  CoreFoundation                  0x3074d37a8         __CFRunLoopDoTimer
13  CoreFoundation                  0x3074d3068         __CFRunLoopDoTimers
14  CoreFoundation                  0x3074d0c8c         __CFRunLoopRun
15  CoreFoundation                  0x307400da4         CFRunLoopRunSpecific
16  S. Contacts                     0x2000fefa8         +[RCTCxxBridge runRunLoop] (RCTCxxBridge.mm:219)
17  Foundation                      0x308b30318         __NSThread__start__
18  libsystem_pthread.dylib         0x3057d368c         _pthread_body
19  libsystem_pthread.dylib         0x3057d359c         _pthread_start

Thread 19 name: 
0   libsystem_kernel.dylib          0x305626e64         __psynch_mutexwait
1   libsystem_pthread.dylib         0x3057d6b8c         _pthread_mutex_lock_wait
2   JavaScriptCore                  0x30f408060         WTF::initializeCurrentThreadInternal(char const*)
3   JavaScriptCore                  0x30f407eec         WTF::threadEntryPoint(void*)
4   JavaScriptCore                  0x30f407e8c         WTF::wtfThreadEntryPoint(void*)
5   libsystem_pthread.dylib         0x3057d368c         _pthread_body
6   libsystem_pthread.dylib         0x3057d359c         _pthread_start



EOF
```

--- a/ios/RNSentry.h
+++ b/ios/RNSentry.h
@@ -3,18 +3,13 @@
 #else
 #import "RCTBridge.h"
 #endif
-#if __has_include(<React/RCTExceptionsManager.h>)
-#import <React/RCTExceptionsManager.h>
-#else
-#import "RCTExceptionsManager.h"
-#endif
 #if __has_include(<React/RCTRootView.h>)
 #import <React/RCTRootView.h>
 #else
 #import "RCTRootView.h"
 #endif
 
-@interface RNSentry : NSObject <RCTBridgeModule, RCTExceptionsManagerDelegate>
+@interface RNSentry : NSObject <RCTBridgeModule>
 
 + (void)installWithRootView:(RCTRootView *)rootView;
 + (void)installWithBridge:(RCTBridge *)bridge;

--- a/ios/RNSentryEventEmitter.m
+++ b/ios/RNSentryEventEmitter.m
@@ -24,6 +24,10 @@ RCT_EXPORT_MODULE();
              };
 }
 
++ (BOOL)requiresMainQueueSetup {
+    return YES;
+}
+
 - (NSArray<NSString *> *)supportedEvents {
     return @[kEventSentSuccessfully, kEventStored, kModuleTable];
 }

